gpiozero/pins/local.py

```diff
@@ -64,7 +66,20 @@ def _get_revision(self):
                     if overvolted:
                         revision = revision[-4:]
                     return revision
-        raise PinUnknownPi('unable to locate Pi revision in /proc/cpuinfo')
+
+        # Fallback to reading /proc/device-tree if available to support 64-bit systems
+        try:
+            with io.open('/proc/device-tree/system/linux,revision', 'rb') as f:
+                f.seek(1)
+                revision = int.from_bytes(f.read(), byteorder='big', signed=False)
+                if revision != 0:
+                    return hex(revision)[2:]
```

Comment:
I wanted to interfere with your existing way how it works as less as possible. But it would make sense of course. However, beside any other improvements I would focus on the test/mocking issue because that's actually blocking. Besides, tests first, refactoring after seems to be the safer approach.